packages/trader/src/App/Components/Elements/Errors/__tests__/error-component.spec.jsx

```diff
@@ -0,0 +1,109 @@
+import React from 'react';
+import { Router } from 'react-router';
+import { createBrowserHistory } from 'history';
+import { render, screen } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+import ErrorComponent from '../error-component';
+
+describe('ErrorComponent', () => {
+    const common_error_message = 'Sorry, an error occured while processing your request.';
+    const test_title = 'Test title';
+    const test_message = 'Test message';
+    const redirect_button = 'Confirm';
+    const default_button = 'Ok';
+    const refresh_button = 'Refresh';
+    const browser_history = createBrowserHistory();
+    const reloadFn = () => {
+        window.location.reload();
+    };
+
+    beforeAll(() => {
+        Object.defineProperty(window, 'location', {
+            configurable: true,
+            value: { reload: jest.fn() },
+        });
+    });
+
+    it('should render ErrorComponent', () => {
+        const { container } = render(
+            <Router history={browser_history}>
+                <ErrorComponent />
+            </Router>
+        );
+
+        expect(container).toBeInTheDocument();
+    });
+    it('should render child component Dialog if ErrorComponent recive is_dialog === true', () => {
+        render(<ErrorComponent is_dialog />);
+
+        expect(screen.getByText('There was an error')).toBeInTheDocument();
+        expect(screen.getByText(default_button)).toBeInTheDocument();
+        expect(screen.getByText(common_error_message)).toBeInTheDocument();
+    });
+    it('should render child component Dialog with recieved title, message and redirect label if ErrorComponent passed them and is_dialog === true', () => {
+        render(
+            <ErrorComponent is_dialog header={test_title} message={test_message} redirect_label={redirect_button} />
+        );
+
+        expect(screen.getByText(test_title)).toBeInTheDocument();
+        expect(screen.getByText(redirect_button)).toBeInTheDocument();
+        expect(screen.getByText(test_message)).toBeInTheDocument();
+    });
+    it('should render PageError component if ErrorComponent recive is_dialog === false', () => {
+        render(
+            <Router history={browser_history}>
+                <ErrorComponent />
+            </Router>
+        );
+
+        expect(screen.getByText('Something’s not right')).toBeInTheDocument();
+        expect(screen.getByText(common_error_message)).toBeInTheDocument();
+        expect(screen.getByText(refresh_button)).toBeInTheDocument();
+    });
+    it('should render PageError component with recieved header, message and redirect label if ErrorComponent passed them and is_dialog === false', () => {
+        render(
+            <Router history={browser_history}>
+                <ErrorComponent header={test_title} message={test_message} redirect_label={redirect_button} />
```

Comment:
```suggestion
                <ErrorComponent header={test_title} message={test_message} redirect_label={redirect_button} />
```
here we are not specifying whether the `is_dialog={false}

Reply:
As `is_dialog` has no default value, then if the component doesn't receive `is_dialog`, then it will be equal to `undefined`. Based on the value of `is_dialog` we render `<Dialog />` or `<PageError />`:
`is_dialog ? <Dialog /> : <PageError />`
So, if `is_dialog === undefined`  we will render `<PageError />` (same as if we would passed `is_dialog = {false}`)